src/interchange/src/json.rs

```diff
@@ -251,7 +258,7 @@ fn build_row_schema_field_type(
     type_namer: &mut Namer,
     custom_names: &BTreeMap<GlobalId, String>,
     typ: &ColumnType,
-    set_null_defaults: bool,
+    options: SchemaJsonOptions,
```

Comment:
We should be able to avoid all these clones by passing a reference here (and in `build_row_schema_fields`) rather than an owned object.

Reply:
I had started with that, but the `item_id` inside it needs to be updated for nested avro records.

Reply:
I guess if I extract out the `item_id` as a separate method parameter that would work.

Reply:
Updated!